corehq/util/files.py

```diff
@@ -59,7 +59,8 @@ class TransientTempfile(object):
     """
 
     def __enter__(self):
-        _, self.path = tempfile.mkstemp()
+        fd, self.path = tempfile.mkstemp()
+        os.close(fd)
```

Comment:
There's a smell here because this is closing, and then inside the context manager immediately re-opening the same file. From a resource management perspective this might be more efficient:

```py
@attr.s
class TransientTempfile(object):
    """
    Manage a temporary file that can be opened and closed as needed, but will
    be automatically cleaned up on exit.

        with TransientTempfile() as fileobj, path:
            # path exists throughout this block
            with fileobj:
                fileobj.write("Adding stuff to the file")
            with open(path) as f:
                do_stuff(f.read())
        # path no longer exists
    """

    mode = attr.ib(default="wt")
    encoding = attr.ib(default="utf-8")

    def __enter__(self):
        fd, self.path = tempfile.mkstemp()
        self.encoding = None if "b" in self.mode else self.encoding
        fileobj = io.open(fd, self.mode, encoding=self.encoding)
        return fileobj, self.path

    def __exit__(self, exc_type, exc_val, exc_tb):
        os.remove(self.path)
```

Maybe this makes the implementation too complex? Your call if you want to use it or not.

Reply:
I thought about making this util manage opening and closing the tempfile as well, but I like the idea of keeping it responsible solely for allocating the tempfile and not making a new api for file access.  I think I'll keep it as-is for now.